generate_model.py

```diff
@@ -0,0 +1,82 @@
+"""Uses the model template to generate a model"""
+
+import json
+import pathlib
+import os
+from uuid import uuid4
+import jinja2
+from pydantic import ValidationError
+from model import ParametersModel
+from helpers import combine_variables
+
+# output-related variables
+experiment_count = 0  # current number of combinations
+template_folder_name = "templates"  # output folder
+output_folder_base = "experiment"  # output folder
+
+# load the templates
+parent_dir = os.path.dirname(__file__)
+templates_folder = os.path.join(parent_dir, template_folder_name)
+environment = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(templates_folder)
+)
+
+# read the variables for rendering
+with open("variables.json", mode="r", encoding="utf-8") as variables_input:
+    model_variables = json.load(variables_input)
+
+# create a pydantic model
+model_parameter_list = ParametersModel(variables=model_variables)
+
+# validate the model
+try:
+    ParametersModel.model_validate(model_parameter_list)
+except ValidationError as e:
+    print(e)
+
+# print(model_parameter_list.model_json_schema())  # show the schema
+print(f"parsed variables: {model_parameter_list.model_dump_json()}\n")  # show the parsed values
+
+# loop over the variable combinations and render all the files
+for parameter_grid_item in combine_variables(model_parameter_list.variables):
+    # define experiment name
+    experiment_name = f"{output_folder_base}_{experiment_count}"
```

Comment:
Maybe we need to remove experiment_n, experiment_n+1, etc with e.g.
```
# Check if the directory exists
    if os.path.isdir(experiment_name):
        # If it exists, remove it
        shutil.rmtree(experiment_name)
```
The only problem with this is that it will run a maximum of n times. If it is less than in a previous run some of the experiments the high number will remain.